packages/jsii-pacmak/lib/targets/python.ts

```diff
@@ -1115,15 +1115,9 @@ class Package {
         for (const depName of Object.keys(expectedDeps)) {
             const depInfo = expectedDeps[depName];
             // We need to figure out what our version range is.
-            // Basically, if it starts with Zero we want to restrict things to
-            // ~=X.Y.Z. If it does not start with zero, then we want to do ~=X.Y,>=X.Y.Z.
+            // If the version is X.Y.Z, we want to allow any compatible version X.Y.n
             const versionParts = depInfo.version.split(".");
-            let versionSpecifier: string;
-            if (versionParts[0] === "0") {
-                versionSpecifier = `~=${versionParts.slice(0, 3).join(".")}`;
-            } else {
-                versionSpecifier = `~=${versionParts.slice(0, 2).join(".")},>=${versionParts.slice(0, 3).join(".")}`;
-            }
+            const versionSpecifier = `~=${versionParts.slice(0, 3).join(".")}`;
```

Comment:
Just to make sure, `~=1.4.0` will be compatible with `1.6.5`, correct?

Reply:
Not totally sure I understand the question.  If you use a version string of "~=1.4.0" for a dependency, it will be compatible with 1.4.*.  If the only version it can find is 1.6.5, it will fail.  Is that not correct?

Reply:
In JavaScript the version spec we use is “^1.3.0” which means to use the latest compatible version. This basically includes all versions >= 1.3.0 and < 2.0.0 since by definition those should be backwards compatible with 1.3.0. We should use the same spec for python I think, no?

Reply:
I thought that was the problem we were trying to fix in https://github.com/aws/aws-cdk/issues/3517.  If I try to install version 1.0.0 of a module and it has dependencies, all of the dependencies come in as the latest available version (e.g. 1.3.0) rather than also being pinned to 1.0.0.

Reply:
As long as 1.3.0 is compatible with 1.0.0, I don’t see a problem.

Reply:
Mmmm... looking at the CDK issue, I am curious how this can happen because basically there is a violation of the requirement for the transitive dependency. If there’s a top level dependency A-1.0.0 and B-1.0.0 and day A has a transitive dependency on C, which takes a dependency on B. If we bring in C@1.3.0 it’s requirement on B would be ^1.3.0 which mean that B is not compatible.

Reply:
Well, from the issue above it appears that it is causing a problem.  Plus, I guess if I pinned my dependency to a specific version my expectation is that all transitive dependencies would be pinned to the same version.  If we don't think that is the right behavior this fix is not needed as there really isn't a problem to fix.

Reply:
I don't think this PR should be merged.  It would mean that Python would work differently than the other languages.  Instead, I will investigate the linked issue and try to reproduce it and figure out why it caused an error.

Reply:
Makes sense